### PR TITLE
feat: DuplicatePeer error

### PR DESCRIPTION
**Pull request by mkurapov**
## Changes proposed in this pull request

- Adds DuplicatePeer error during peering creation via checking for new unique constraint/index in the DB

## Context

While building out the auto-peering service, I noticed that it didn't really make sense to allow creating a Peer with the same combination of `staticIlpAddress` and `assetId` (this would mean a duplicate peering relationship). This adds a unique constraint for these two columns in the DB.

Related to #1761 

## Checklist

- [ ] Related issues linked using `fixes #number`
- [x] Tests added/updated
- [ ] Documentation added
- [ ] Make sure that all checks pass
- [ ] Postman collection updated

### CODE REVIEW

File: packages/backend/src/peer/service.test.ts [hunk 1]
```diff
@@ -173,6 +173,21 @@ describe('Peer Service', (): void => {
...
+      const peer = await peerService.create(options)
+      assert.ok(!isPeerError(peer))
+
+      await expect(
+        peerService.create({
+          ...options,
+          staticIlpAddress: options.staticIlpAddress,
+          assetId: options.assetId
+        })
+      ).resolves.toEqual(PeerError.DuplicatePeer)
```
Comment (BlairCurrey) [inline packages/backend/src/peer/service.test.ts:189]:
```suggestion
      await expect(
        peerService.create(options)
      ).resolves.toEqual(PeerError.DuplicatePeer)
```

**Review by BlairCurrey — APPROVED** [review]
Left a suggestion but looks good. Tested on postman as well and see the 409 duplicate peer error.

File: packages/backend/src/peer/errors.ts [hunk 2]
```diff
@@ -24,6 +26,7 @@ export const errorToMessage: {
   [key in PeerError]: string
 } = {
   [PeerError.DuplicateIncomingToken]: 'duplicate incoming token',
+  [PeerError.DuplicatePeer]: 'duplicate peer',
```
Comment (raducristianpopa) [inline packages/backend/src/peer/errors.ts:29]: Would it be better to provide a more specific error message rather than simply "Duplicate peer"?

Comment (mkurapov) [reply packages/backend/src/peer/errors.ts]: Updated to be more specific

**Review by sabineschaller — APPROVED** [review]
Thanks for making this so digestible.

**Review by raducristianpopa — APPROVED** [review]

**Pull request merged**